WaterLogged.Supplement/EmailListener.cs

```diff
@@ -0,0 +1,119 @@
+﻿using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Net.Mail;
+
+namespace WaterLogged.Supplement
+{
+    //TODO: 
+    //Add throttling to the write. To allow this to support a large number of instances/apps, sending them in batches every 30 seconds, or whatever user sets
+    //would be a nice feature add. Background task and collection of events pushed into one message at the interval would probably be best.
+
+    /// <summary>
+    /// A class to send logging updates via email. 
+    /// </summary>
+    public class EmailListener : Listener
+    {        
+        public string UserName { get; set; }
+        public string UserPass { get; set; }
+        public int Port { get; set; }
+        public bool SSL { get; set; }
+        public string EmailSubject { get; set; }  
+        public string SMTPHost { get; set; }
+        public string FromAddress { get; set; }
+
+        //A hacky way to have a threadsafe collection with no duplicates, get it together C#.
+        private ConcurrentDictionary<string, byte> _recipients;
+
+        /// <summary>
+        /// Contruct the email listener with host, port, security, netwrok crednetials, and a list of reciepients
+        /// </summary>
+        /// <param name="host">Location of the SMPT Server</param>        
+        /// <param name="ssl">Use an SSL connection</param>
+        /// <param name="userName">user name for netwrok credentials</param>
+        /// <param name="userPass">Password for network credentials</param>
+        /// <param name="recipients">A list of recipients to recieve email alerts</param>
+        /// <param name="port">Port for the smpt server, defaults to 587</param>
+        /// <param name="emailSubject">Email message subject</param>
+        public EmailListener(string host, string fromAddress, bool ssl = false, int port = 587, string emailSubject = "WaterLogged Message", 
```

Comment:
You used cascading constructors before, but this is what I normally do. I am not tied to either way, feel free to change. All the same really.

Reply:
Nah, it's good. Maybe if this library actually gets successful I'll impose some kind of standard. But they won't be *my* standards. By that time hopefully a couple of people have contributed and I can try to pull out what's common between them.